woocommerce/woocommerce-ios
# Shipping Labels M1: UI to refund a shipping label

Fixes #2168 

## Why

After the backend support https://github.com/woocommerce/woocommerce-ios/pull/3294, this PR implements the UI integration to request a refund for a shipping label.

## Changes

- Created `RefundShippingLabelViewController` and view model `RefundShippingLabelViewModel` to display refund information with a CTA to request a refund
- Created `PlainTextSectionHeaderView` for displaying plain text in a table view section header. The reasons I didn't create a xib:
  - The view only contains one `UILabel` and the logic was pretty straightforward
  - There is no option to create a xib when creating a `UITableViewHeaderFooterView`
- In `OrderDetailsViewController`, integrated with `RefundShippingLabelViewController` when the user taps on the refund request CTA

## Testing

Prerequisites: the site has [WooCommerce Shipping & Tax plugin](https://docs.woocommerce.com/document/woocommerce-shipping-and-tax/). You can set up your store to use a test credit card with instructions in p91TBi-3xD-p2.

Please note that in testing mode, a shipping label refund only works if it is requested **shortly** after label creation.

- Go to the orders tab, open an order with at least one physical product
- In wp-admin, create a shipping label for the order above
- Right after the shipping label is created, go back to the app and pull down to refresh the order details --> a shipping label package should be shown after syncing
- Tap on the ellipsis in the package card section header, then tap "Request a Refund" --> a screen should be shown with purchase date and refundable amount, plus a CTA at the bottom
- Tap "Refund Label ({refundable_amount})" CTA --> it should navigate back to order details, then a success snackbar is shown when the refund is requested. In the meantime, the package card should be updated to refunded state (no order items shown)

## Example screenshots

![simulator](https://user-images.githubusercontent.com/1945542/101447118-8ae6f400-395f-11eb-8a43-f57bb86237ea.gif)

portrait | landscape
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 13 46 17](https://user-images.githubusercontent.com/1945542/101447191-ace07680-395f-11eb-82f1-79e263e16a03.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 13 46 40](https://user-images.githubusercontent.com/1945542/101447198-b073fd80-395f-11eb-8802-22fb082295b3.png)
![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 13 47 20](https://user-images.githubusercontent.com/1945542/101447200-b10c9400-395f-11eb-8a06-d85832b41f6f.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 13 47 24](https://user-images.githubusercontent.com/1945542/101447201-b1a52a80-395f-11eb-83b7-ab5274c44877.png)

Please notice the snackbar at the bottom:

error state | success state
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 13 47 03](https://user-images.githubusercontent.com/1945542/101447256-ce416280-395f-11eb-8424-335011049504.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 14 04 53](https://user-images.githubusercontent.com/1945542/101447263-d1d4e980-395f-11eb-861a-7515fca4136f.png)


Update release notes:
